knotx-core/src/main/java/io/knotx/dataobjects/Fragment.java

```diff
@@ -102,7 +145,45 @@ public JsonObject context() {
    * @return <tt>true</tt> if this Fragment has no dynamic parts (contains only static markup).
    */
   public boolean isRaw() {
-    return knots.contains(RAW_FRAGMENT_ID);
+    return knots().contains(RAW_FRAGMENT_ID);
+  }
+
+  public boolean isFallback() {
+    return knots().contains(FALLBACK_FRAGMENT_ID);
+  }
+
+  /**
+   * @return true if processing of this Fragment has failed
+   */
+  public boolean failed() {
+    return this.knots.stream().anyMatch(k -> k.getStatus() == KnotTaskStatus.FAILURE);
+  }
+
+  /**
+   * @return replacement markup that should be rendered if this Fragment has failed. Can be empty
+   * string. Null value indicates that no replacement markup is provided.
```

Comment:
Could you please fix javadoc, this never should be null. You probably ment:
> absent value indicates that no replacement markup is provided.

Reply:
Fixed